lib/Convos/Core/User.pm

```diff
@@ -151,21 +155,11 @@ sub validate_password {
   my ($self, $plain) = @_;
 
   return 0 unless $self->password and $plain;
-  return 1 if $self->_bcrypt($plain, $self->password) eq $self->password;
+  return 1 if $self->_crypt->verify_password($plain, $self->password);
+  # XXX: Should this also upgrade hash if $self->_crypt->needs_rehash($plain) ?
```

Comment:
I think it would be a great addition if the password would get upgraded automatically.

Reply:
This is fixed.

(Not sure if calling save_p is neccecary?)